src/components/flat-table/flat-table.component.tsx

```diff
@@ -203,12 +203,23 @@ export const FlatTable = ({
 
   useLayoutEffect(() => {
     const findSelectedId = () => {
-      const firstfocusableElement = tableRef.current?.querySelector(
-        FOCUSABLE_ROW_AND_CELL_QUERY
+      const focusableElements = Array.from(
+        tableRef.current?.querySelectorAll(FOCUSABLE_ROW_AND_CELL_QUERY) ||
+          /* istanbul ignore next */ []
       );
 
+      if (
+        !focusableElements.length ||
```

Comment:
**suggestion**: I don't think you need the `length` check, empty arrays will never pass a `.some` check (they don't have any elements at all so can't have any elements that pass a particular test). See here from the browser console even with an "autopass" check:

![image](https://github.com/Sage/carbon/assets/25186376/2a0e11af-7974-4799-b4f1-b86e6f2218be)